deploy_snapshot.sh

```diff
@@ -21,28 +21,80 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
-if [ "$TRAVIS_REPO_SLUG" != "OfficeDev/ews-java-api" ]; then 
-	echo "[DEPLOY] Skipping snapshot deployment for repo:'$TRAVIS_REPO_SLUG'."
+set -e # Any subsequent(*) commands which fail will cause the shell script to exit immediately
+
+if [ "$TRAVIS_REPO_SLUG" != "OfficeDev/ews-java-api" ]; then
+    echo "[DEPLOY] Skipping snapshot deployment for repo:'$TRAVIS_REPO_SLUG'."
 elif [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
-	echo "[DEPLOY] Skipping snapshot deployment for a pull request."
+    echo "[DEPLOY] Skipping snapshot deployment for a pull request."
 elif [ "$TRAVIS_BRANCH" != "master" ]; then
-	echo "[DEPLOY] Skipping snapshot deployment for branch:'$TRAVIS_BRANCH'."
+    echo "[DEPLOY] Skipping snapshot deployment for branch:'$TRAVIS_BRANCH'."
 elif [ "$TRAVIS_SECURE_ENV_VARS" == "false" ]; then
-	echo "[DEPLOY] Skipping snapshot deployment due to TRAVIS_SECURE_ENV_VARS is set to '$TRAVIS_SECURE_ENV_VARS'."
+    echo "[DEPLOY] Skipping snapshot deployment due to TRAVIS_SECURE_ENV_VARS is set to '$TRAVIS_SECURE_ENV_VARS'."
 elif [ "$TRAVIS_JDK_VERSION" != "oraclejdk7" ]; then
-	echo "[DEPLOY] Skipping snapshot deployment for jdk:'$TRAVIS_JDK_VERSION'."
-else 
-	echo "[DEPLOY] Deploying snapshot for commit:'$TRAVIS_COMMIT' @ build-id:'$TRAVIS_BUILD_ID'"	
-	# create settings.xml
-	echo "<settings><servers><server><id>ossrh-snapshot</id><username>${OSSRHUSER}</username><password>${OSSRHPASS}</password></server></servers></settings>" > $HOME/.m2/settings.xml
-	# deploy
-	if [ -z "${GPG_PASSPHRASE+xxx}" ]; then
-	    echo "[INFO] Deploying unsigned artifacts"
-            mvn clean deploy --settings="$HOME/.m2/settings.xml" -Dmaven.test.skip=true
-        else
-            echo "[INFO] Deploying signed artifacts"
-            mvn clean deploy --settings="$HOME/.m2/settings.xml" -Dmaven.test.skip=true -Dgpg.passphrase=$GPG_PASSPHRASE
-        fi
-	# clean up
-	rm -f $HOME/.m2/settings.xml
-fi
+    echo "[DEPLOY] Skipping snapshot deployment for jdk:'$TRAVIS_JDK_VERSION'."
+else
+    echo "[DEPLOY] Deploying snapshot for commit:'$TRAVIS_COMMIT' @ build-id:'$TRAVIS_BUILD_ID'"
+    # create settings.xml
+    echo "<settings><servers><server><id>ossrh-snapshot</id><username>${OSSRHUSER}</username><password>${OSSRHPASS}</password></server></servers></settings>" > $HOME/.m2/settings.xml
+    # deploy
+    if [ -z "${GPG_PASSPHRASE+xxx}" ]; then
+       echo "[INFO] Deploying unsigned artifacts"
+       mvn clean deploy --settings="$HOME/.m2/settings.xml" -Dmaven.test.skip=true
+    else
+       echo "[INFO] Deploying signed artifacts"
+       mvn clean deploy --settings="$HOME/.m2/settings.xml" -Dmaven.test.skip=true -Dgpg.passphrase=$GPG_PASSPHRASE
+    fi
+    # clean up
+    rm -f $HOME/.m2/settings.xml
```

Comment:
What happens if the script fails when calling mvn and exists.  Are we doing this as an extra step but travis cleans up anyway?  Should we not let the script exist on failure between persisting and deleting the secret?